src/cryptonote_core/tx_pool.cpp

```diff
@@ -147,8 +157,19 @@ namespace cryptonote
           continue;
         }
 
-        if (state_change == pool_tx_state_change)
-          return true;
+        crypto::public_key service_node_to_change_in_the_pool;
+        if (service_node_list.try_resolve_pubkey_in_quorum(quorum_type, quorum_group, pool_tx_state_change.block_height, pool_tx_state_change.service_node_index, service_node_to_change_in_the_pool))
+        {
+          if (service_node_to_change == service_node_to_change_in_the_pool)
+            return true;
+        }
+        else
+        {
+          MERROR("Could not resolve the service node public key from the information in a pooled tx state change, possibly corrupt tx in your blockchain, falling back to primitive checking method");
```

Comment:
It's still possible to have an old tx in the pool if the conflicting txes arrives to you and to a mining pool in a different order (and the other one gets mined first).  I think at most this should be a warning, not an error.

Reply:
I've changed to warning but I don't make the connection here. This is in the case the get_pubkey function fails i.e. you're missing a quorum (or invalid state change- this one shouldn't happen since its already been prevalidated).  Conflicting txes doesn't make this code path trigger in by itself?